client/src/app/shared/services/collaboration.service.ts

```diff
@@ -0,0 +1,147 @@
+    if (!courseIterationId) {
+      return;
+    }
+    const serverCollaborationData = await this.discover(courseIterationId);
+    if (!serverCollaborationData || !serverCollaborationData.allocations) {
+      this.subscribe(courseIterationId);
+      return;
+    }
+
+    if (
+      this.allocationsService.equalsCurrentAllocations(serverCollaborationData.allocations) &&
+      this.constraintsService.equalsCurrentConstraints(serverCollaborationData.constraints) &&
+      this.lockedStudentsService.equalsCurrentLockedStudentsUsingKeyValuePair(serverCollaborationData.lockedStudents)
+    ) {
+      this.subscribe(courseIterationId);
+      return;
+    }
+
+    const overlayData = {
+      title: 'Connected to Collaboration Service',
+      description:
+        'The Collaboration Service has a different allocations and constraints state available. Do you want to load it? This will overwrite your current allocations, constraints and locked students. Not loading it will overwrite the other data. Be careful, this action cannot be undone.',
+      primaryText: 'Use Collaboration Data',
+      primaryButtonStyle: 'btn-secondary',
+      primaryAction: async () => {
+        const serverCollaborationData = await this.discover(courseIterationId);
+        this.allocationsService.setAllocations(serverCollaborationData.allocations, false);
+        this.constraintsService.setConstraints(serverCollaborationData.constraints, false);
+        this.lockedStudentsService.setLocksAsArray(serverCollaborationData.lockedStudents, false);
+        await this.subscribe(courseIterationId);
+        this.overlayService.closeOverlay();
+      },
+      secondaryText: 'Overwrite Collaboration Data',
+      secondaryButtonStyle: 'btn-warn',
+      secondaryAction: async () => {
+        await this.subscribe(courseIterationId);
+        this.overlayService.closeOverlay();
+      },
+      isDismissable: false,
+    };
+
+    this.overlayService.displayComponent(ConfirmationOverlayComponent, overlayData);
+  }
+
+  async disconnect(): Promise<void> {
+    this.discoverySubscription?.unsubscribe();
+    for (const subscription of this.subscriptions || []) {
+      subscription.unsubscribe();
+    }
+    this.subscriptions = [];
+
+    this.websocketService.connection.disconnect();
+  }
+
+  private async subscribeToAllocations(courseIterationId: string): Promise<void> {
+    this.websocketService.send(courseIterationId, 'allocations', this.allocationsService.getAllocationsAsString());
+
+    this.subscriptions.push(
+      await this.websocketService.subscribe(courseIterationId, 'allocations', allocations => {
+        console.log('Received allocations');
+        this.allocationsService.setAllocations(allocations, false);
+      })
+    );
+  }
+
+  private async subscribeToLockedStudents(courseIterationId: string): Promise<void> {
+    this.websocketService.send(courseIterationId, 'lockedStudents', this.lockedStudentsService.getLocksAsString());
```

Comment:
We could think about adding a global constant for lockedStudents (also constraints, etc.) since we are using the strings in multiple files. So we can hopefully avoid errors, if someone wants to change the strings for whatever reason in the future.